airflow/datasets/manager.py

```diff
@@ -52,7 +52,7 @@ def register_dataset_change(
         For local datasets, look them up, record the dataset event, queue dagruns, and broadcast
         the dataset event
         """
-        dataset_model = session.query(DatasetModel).filter(DatasetModel.uri == dataset.uri).one_or_none()
+        dataset_model = session.scalar(select(DatasetModel).where(DatasetModel.uri == dataset.uri))
```

Comment:
what is the benefit of using `scalar_one_or_none()` here?

Reply:
Just to keep the same logic as original query, however might be there is no possible to get more than 1 record here

Reply:
okay, will make it as `session.scalars(...).one()` to be more close to original query.

Reply:
test are failing if I make change according to your suggestion , so retaining my change

Reply:
Just wondering, what kind of error happen?

Reply:
I guess it failed because suggested changes also required to change in mocking object

https://github.com/apache/airflow/blob/eacdeb4ed967313ba49721944c95a3721b5d7afb/tests/datasets/test_manager.py#L57 

```python
mock_session.execute.return_value.scalar_one_or_none.return_value = None
```

In this case locally `TestDatasetManager` executed successfully
![image](https://github.com/apache/airflow/assets/3998685/f7de0e7d-ba17-4359-a816-f39c09326078)

Reply:
Anyway, I've checked locally that if I tried to put with same `DatasetModel.uri` I've got an error

```console
sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) UNIQUE constraint failed: dataset.uri
```

So I guess we don't need `scalar_one_or_none()` in the current codebase.

Reply:
thank you for confirming !